### PR TITLE
feat: _V3 engine api skeletons

### PR DESCRIPTION
**Pull request by mattsse**
ref #2893

not announced via caps yet, but adds the definitions

### CODE REVIEW

**Review by Rjected — COMMENTED** [review]
just wondering about `beaconBlockRoot`, I think we'll also need followups to add some of the new fields to other engine types

File: crates/rpc/rpc-api/src/engine.rs [hunk 1]
```diff
@@ -21,6 +21,14 @@ pub trait EngineApi {
     #[method(name = "newPayloadV2")]
     async fn new_payload_v2(&self, payload: ExecutionPayload) -> RpcResult<PayloadStatus>;
 
+    /// Post Cancun payload handler
+    #[method(name = "newPayloadV3")]
+    async fn new_payload_v3(
+        &self,
+        payload: ExecutionPayload,
+        versioned_hashes: Vec<H256>,
+    ) -> RpcResult<PayloadStatus>;
```
[the remark this answers is not in the document]
Comment (mattsse) [reply crates/rpc/rpc-api/src/engine.rs:30]: nice find, was only looking at

https://github.com/ethereum/go-ethereum/blob/2274a03e339f213361453590b54917bbfd0a0c31/eth/catalyst/api.go#L448-L449

File: crates/rpc/rpc-api/src/engine.rs [hunk 2]
```diff
@@ -21,6 +21,14 @@ pub trait EngineApi {
     #[method(name = "newPayloadV2")]
     async fn new_payload_v2(&self, payload: ExecutionPayload) -> RpcResult<PayloadStatus>;
 
+    /// Post Cancun payload handler
+    #[method(name = "newPayloadV3")]
+    async fn new_payload_v3(
+        &self,
+        payload: ExecutionPayload,
```
[the remark this answers is not in the document]
Comment (Rjected) [reply crates/rpc/rpc-api/src/engine.rs:28]: makes sense

**Review by Rjected — APPROVED** [review]
LGTM

**Comment by mattsse** [comment]
added missing arg

just realized there's a cancun readme, but will do the remaining functions separately

**Review by onbjerg — APPROVED** [review]

**Pull request merged**